ckanext/canada/templates/ajax_snippets/api_tracking_info.html

```diff
@@ -0,0 +1,56 @@
+{#
+Displays information about the payload of a tracke API call.
+
+id - The id of the API tracking object.
+embedded - If true will not include the "modal" classes on the snippet.
+
+Example
+
+    {% snippet 'ajax_snippets/api_tracking_info.html', id=id, embedded=true %}
+
+#}
+
+{% set id = h.sanitize_id(id) %}
+{% if id and h.check_access('sysadmin') %}
+  {% set api_tracking_info = h.canada_api_tracking_info(id) %}
```

Comment:
template helpers are normally used for small bits of formatting code, not code that fetches a bunch of information from the DB. You probably used a helper just because that was your only option with ajax snippets, but that's because ajax snippets are terrible https://github.com/ckan/ckan/issues/7619

It would be much better to create a normal flask view and a new action to fetch the data. That way the permission checking can be done in the normal ckan way in the action instead of a template if statement.

Reply:
yeah that would be better. I still want to have it in an ajax modal however :/ as I think it would be not great to have to go fetch and decrypt all payloads in the sysadmin view. That way, we would only fetch and decrypt on user request

Reply:
There's nothing preventing us from ajax-ing in the response to a normal flask view.